viziphant/unitary_event_analysis.py

```diff
@@ -0,0 +1,401 @@
+    center_of_analysis_window = t_winpos + window_size / 2.
+    n_trials = len(data)
+    joint_surprise_significance = ue.jointJ(significance_level)
+    xlim_left = (min(t_winpos)).magnitude
+    xlim_right = (max(t_winpos) + window_size).magnitude
+
+    if len(params_dict['unit_real_ids']) != n_neurons:
+        raise ValueError(
+            'length of unit_ids should be equal to number of neurons! \n'
+            'Unit_Ids: ' + params_dict['unit_real_ids']
+            + 'not equal number of neurons: ' + n_neurons)
+
+    plt.figure(num=1, figsize=params_dict['figsize'])
+    plt.subplots_adjust(hspace=params_dict['hspace'],
+                        wspace=params_dict['wspace'],
+                        top=params_dict['top'],
+                        bottom=params_dict['bottom'],
+                        left=params_dict['left'],
+                        right=params_dict['right'])
+
+    # set y-axis for raster plots with ticks and labels
+    y_tick_interval = params_dict['y_tick_interval']
+    y_ticks_list = []
+    # find start y-tick position for each neuron trail-set
+    y_ticks_list.extend(range(1, n_neurons * n_trials, n_trials + 1))
+    # find in-between y-tick position for each trail with the specific interval
+    for n in range(n_neurons):
+        in_between_ytick_position = range(n * (n_trials + 1) + y_tick_interval,
+                                          (n + 1) * n_trials, y_tick_interval)
+        y_ticks_list.extend(in_between_ytick_position)
+    y_ticks_list.sort()
+
+    y_ticks_labels_list = [1]
+    number_of_in_between_y_ticks_per_neuron = math.floor(
+        n_trials / y_tick_interval)
+    y_ticks_labels_list = \
+        np.r_[1, np.arange(1, number_of_in_between_y_ticks_per_neuron + 1)
+              * y_tick_interval]
+
+    # adding n_neuron times the y_ticks_labels_list to itself, so that each
+    # neuron has the same y_ticks_labels
+    y_ticks_labels_list = np.tile(y_ticks_labels_list, n_neurons)
+
+    def mark_epochs(axes_name):
+        """
+        Marks epochs on the respective axis by creating a vertical line and
+        shows the epoch's name under the last subplot. Epochs need to be
+        defined in the plot_params_user dictionary.
+        Parameters
+        ----------
+        axes_name : matplotlib.axes._subplots.AxesSubplot
+            The axis on which the epochs will be marked.
+        """
+        for key in params_dict['events'].keys():
+            for event_timepoint in params_dict['events'][key]:
+                axes_name.axvline(event_timepoint, ls='-',
+                                  lw=params_dict['lw'], color='r')
+                if axes_name.get_geometry()[2] == 6:
+                    axes_name.text(x=event_timepoint-10*pq.ms, y=-90, s=key,
+                                   fontsize=12, color='b')
+
+    print('plotting Unitary Event Analysis ...')
+
+    print('plotting Spike Events ...')
+    axis1 = plt.subplot(6, 1, 1)
```

Comment:
It should be `axes` not `axis`. Axes contain figure elements. Axis contains ticks, labels, etc.. Please change to `ax` or `axes` throughout.
See here: https://matplotlib.org/1.5.1/faq/usage_faq.html#parts-of-a-figure